### PR TITLE
Drop support for Ruby < 2.6 and JRuby < 9.3

### PR DESCRIPTION
**Pull request by johnnyshields**
Fixes https://github.com/onelogin/ruby-saml/issues/608

This PR removes support for Ruby < 2.1 and JRuby < jruby-9.1

### CODE REVIEW

**Comment by johnnyshields** [comment]
@pitbulk ready for your review

**Comment by pitbulk** [comment]
I want to maintain those versions supported for now, make a new release with all the changes that were introduced recently and let's see if that gonna be the latest version for those deprecated ruby versions.

I will keep the PR open to use it "soon"

**Comment by johnnyshields** [comment]
@pitbulk let's merge this one soon?

**Comment by petergoldstein** [comment]
@pitbulk @johnnyshields So given the recent ownership changes for the repo, and the corresponding change in maintenance burden I'd suggest both that this be merged and that a more restrictive view to supported Ruby versions be taken going forward.  Supporting old, deprecated Ruby versions tends to be VERY high maintenance.

One approach might be:

1. Cut a release that includes support for existing Rubies (">= 1.8.7" according to the gemspec) and has all current updates
2. Drop everything from CI/gemspec/etc. except for Rubies that are still maintained (MRI 2.7+, JRuby, Truffleruby)
3. Update the code base to reflect (2), allowing a degree of code modernization.  Rubocop may be helpful here
4. Release a go-forward version with a new major version number to reflect these changes.

This would make the gem much easier to maintain going forward, ensure those on older Rubies have a release that incorporates recent changes, and support security patches for the older version if necessary.

Thoughts?

**Comment by johnnyshields** [comment]
@petergoldstein that sounds like a good plan to me.

**Comment by pitbulk** [comment]
Yes, that seems to be a good plan

**Comment by petergoldstein** [comment]
@pitbulk And if you need any help maintaining this gem, please let me know if I can be of assistance.  I'm happy to help.

**Comment by pitbulk** [comment]
Ok, I created v2.x branch, let's work on this branch on everything related to old ruby version related.
Once ready, we can publish an official release

**Comment by johnnyshields** [comment]
@pitbulk please run CI and merge. It would be nice if you can add me to auto-run the CI from the next time.

File: Gemfile [hunk 1]
```diff
@@ -4,3 +4,12 @@
...
 gemspec
+
+gem 'minitest', '~> 5.5'
+gem 'mocha', '~> 0.14'
+gem 'rake', '>= 12.3.3'
+gem 'shoulda', '~> 2.11'
+gem 'simplecov', '< 0.22.0'
+gem 'simplecov-lcov', '> 0.7.0'
+gem 'systemu', '~> 2'
+gem 'timecop', '~> 0.9'
```
Comment (johnnyshields) [inline Gemfile:15]: It is a best practice now (according to Rubocop) to move development dependencies out of the `gemspec` and into the `Gemfile`. Note the `Gemfile` is only for development not loaded when users include the gem in a project.

**Comment by pitbulk** [comment]
I sent you a "triager collaborator invitation", I hope that will allow you to run CI yourself.

**Comment by StoneCypher** [comment]
it looks like your test rig changed.

probable cause is [source line 7](https://github.com/SAML-Toolkits/ruby-saml/actions/runs/4338525755/jobs/7592505019#step:5:8)

> *** Mocha deprecation warning: Change `require 'mocha'` to `require 'mocha/setup'`.

&nbsp;

&nbsp;

and then the actual triggering failure:

[source line 12](https://github.com/SAML-Toolkits/ruby-saml/actions/runs/4338525755/jobs/7592505019#step:5:13)

> /opt/hostedtoolcache/Ruby/2.6.6/x64/lib/ruby/gems/2.6.0/gems/shoulda-2.11.3/lib/shoulda/integrations/test_unit.rb:1:in `require': cannot load such file -- test/unit (LoadError)

you also have missing instance variables inbetween

i don't speak ruby, but, my intuition is that the way to pull the test rig changed, and once you change that and re-set the instance variables, these will all go back to working

&nbsp;

&nbsp;

[you can see someone else going through this here](https://github.com/freerange/mocha/issues/107)

**Comment by johnnyshields** [comment]
@StoneCypher thanks! @pitbulk can you please run tests again?

**Comment by johnnyshields** [comment]
@pitbulk Tests are passing and this is now ready to merge.

File: ruby-saml.gemspec [hunk 2]
```diff
@@ -20,85 +20,9 @@ Gem::Specification.new do |s|
...
-  elsif RUBY_VERSION < '2.0'
-    # 1.9.x
-    s.add_development_dependency('debugger-linecache', '~> 1.2.0')
-    s.add_development_dependency('debugger', '~> 1.6.4')
-  elsif RUBY_VERSION < '2.1'
-    # 2.0.x
-    s.add_development_dependency('byebug', '~> 2.1.1')
-  else
-    # 2.1.x, 2.2.x
-    s.add_development_dependency('pry-byebug')
```
Comment (pitbulk) [inline ruby-saml.gemspec:102]: should we move pry-debug to Gemfile as well?

Comment (johnnyshields) [reply ruby-saml.gemspec:102]: Nah let's just get rid of it. People can add it on their own when doing development. I tend to think the development gems in the Gemfile should be the minimal set which is actually required to get the tests to pass.

Comment (pitbulk) [reply ruby-saml.gemspec:102]: Ok, I will merge after that then

**Pull request merged**